### PR TITLE
Initial iOS dependency structure, added arrow for iOS

### PR DESCRIPTION
**Pull request by ilijapuaca**
- I opted for splitting these up on per-architecture folders instead of a single binary in order to add more flexibility, as we'll be able to cross compile our own library on per-architecture basis
- Added a detailed writeup on issues/workarounds for building arrow. I plan on posting these to their dev list once that markdown is merged
- Each of the architectures contains its own `include` folder, just like the current setup. That's a lot of duplicated files, perhaps we should think about how to minimize these?

Files needing review are `README.md` and `iOS-BUILD.md`, rest of the files are checked in artifacts

### CODE REVIEW

**Comment by isaacbrodsky** [comment]
> Files needing review are `README.md` and `iOS-BUILD.md`, rest of the files are checked in artifacts

For the future, it may be a good idea to make those changes in a separate commit from the artifacts themselves. Then that commit can be looked at in isolation.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
-| x64 | Linux (Ubuntu Bionic)
+| CPU    	| Operating System      	| Notes          	|
+|--------	|-----------------------	|----------------	|
+| x64    	| Mac OSX               	|                	|
+| x64    	| Linux (Ubuntu Bionic) 	|                	|
+| armv7  	| iOS                   	| Built Manually 	|
+| armv7s 	| iOS                   	| Built Manually 	|
+| arm64  	| iOS                   	| Built Manually 	|
+| arm64e 	| iOS                   	| Built Manually 	|
+| x86_64 	| iOS                   	| Built Manually 	|
```
Comment (isaacbrodsky) [inline README.md:15]: nit: I would standardize how we write x64/x86_64/amd64 in this table, just to make scanning it easier

**Comment by ilijapuaca** [comment]
> May be useful to add an explicit set of steps to follow to resolve the current build issues. If those are already being worked on CMake and we expect updates from them to fix this, maybe it's not needed.

Detailed set of steps is in `docs/iOS-BUILD.md`

**Comment by isaacbrodsky** [comment]
> > May be useful to add an explicit set of steps to follow to resolve the current build issues. If those are already being worked on CMake and we expect updates from them to fix this, maybe it's not needed.
> 
> Detailed set of steps is in `docs/iOS-BUILD.md`

Yes, what I was referring to is a walkthrough of steps to follow, so it can be done in one go, rather than addressing build errors as they appear. This is also an OK approach since it lets us remove steps as the build errors no longer happen.

File: README.md (hunk 1 above)
Comment (ilijapuaca) [reply README.md:15]: I was on the fence about this one, as standardizing these may introduce some confusion as it wouldn't match parameters specified when building appropriate architectures (`vcpkg` vs `cmake`)

**Pull request merged**